core/views.py

```diff
@@ -0,0 +1,53 @@
+from django.db.models.fields import BooleanField
+from django.shortcuts import render, get_object_or_404, redirect
+from .models import Habit, Record
+from django.utils import timezone
+from django.contrib.auth.decorators import login_required
+from .forms import HabitForm, HabitRecordForm
+
+# Create your views here.
+def homepage(request):
+    return render(request, "habittracker/homepage.html")
+
+@login_required
+def profile_page(request):
+    return render(request, "habittracker/profile_page.html")
+
+@login_required
+def list_habit(request):
+    habit = Habit.objects.all()
```

Comment:
Your query is getting all the habits but your variable here, that you use in the view context, is singular. Your template expects a list of all habits, right?